docs/faq/about-map.md

```diff
@@ -0,0 +1,387 @@
+---
+title: What's All This About Map Then?
+description: ''
+---
+
+## Caveats and Disclaimers
+
+I make no guarantees that the code examples compile or execute. All code examples are syntactically similar to Typescript, but should be treated as pseudo-code. Also, I expect to reference behaviors (methods, etc) that actually don't exist in the JavaScript standard library (though it'd be nice if some of them did). I try to call out where I'm doing this, but really who knows if I got every case. The intent of this is not because I'm lazy (well, not exclusively because I'm lazy), but because I want to emphasize the concepts, and not muddy the waters of learning with the intricacies of the JavaScript implementation reality.
+
+With that out of the way, on to the topic at hand...
+
+## But First... (a Pure Function Primer)
+
+Before we tackle Map and AndThen, we should make sure that we are all on the same page about functions.
+
+In the mathematical sense, a function is just a mapping from a Set of inputs to a Set of outputs. The output for any given input will always be the same. In the programming world, we call these functions "pure" functions. and they are becoming quite popular.
+
+Pure functions are defined by having no side effects and always returning the same value for the same arguments.This idea is often called "referential integrity", which simply means that if you replace your function call with the value returned from the function, your program doesn't break.
+
+Let's take a moment to explore referential integrity. Given this function:
+
+### add2
+
+        const add2 = (n: number) => n + 2;
```

Comment:
In most places we wrap our code blocks in 

````
```typescript 
// code here
```
````

Could we do that in this file too?

````suggestion
```typescript
const add2 = (n: number) => n + 2;
```
````

Reply:
I was wondering how we had syntax highlighting, thanks!! 😁